src/btree/row_srch.c

```diff
@@ -116,6 +127,42 @@ __wt_search_insert(
             last_ins = ins;
             key.data = WT_INSERT_KEY(ins);
             key.size = WT_INSERT_KEY_SIZE(ins);
+            /*
+             * We have an optimization to reduce the number of bytes we need to compare during the
+             * search if we know a prefix of the search key matches the keys we have already
+             * compared on the upper stacks. This works because we know the keys become denser down
+             * the stack.
+             *
+             * However, things become tricky if we have another key inserted concurrently to next of
```

Comment:
> However, things become tricky if we have another key ... we may wrongly skip the comparison ... and land on the wrong spot.

I think this framing is backwards. It suggests that we update the pointers in the stack in order to avoid breaking  the prefix optimization.

I would describe this by saying that the insert is designed so any concurrent reader will alway see a valid skip list. I.e., at any level, of the list, keys are in sorted order, and that if a reader sees a key in level N, that key is also in levels below N. 

In this framing, the WT-10461 bug is an example of how code (the prefix optimization) that depends on seeing a valid skip list can break if that guarantee is violated.

The value of describing it this way is that it makes it clear the skip list invariants are required independent of the prefix optimization.

Reply:
This is a good idea.